packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -113,6 +117,18 @@ class HotRunner extends ResidentRunner {
     for (FlutterDevice device in flutterDevices)
       device.initLogReader();
 
+    if (previewDart2 && (generator == null)) {
+      generator = new ResidentCompiler(
+        artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath));
+      final String outputFile = await generator.compile(mainPath);
+      // If compilation was successful, no reason VM should reject it.
```

Comment:
is that a safe assumption? People run `flutter run` with bad code sometimes, or change the code on disk between gradle's work and the app loading on the device.